nvflare/app_common/workflows/cross_site_eval.py

```diff
@@ -0,0 +1,187 @@
+    ):
+        super().__init__(*args, **kwargs)
+
+        self._cross_val_dir = cross_val_dir
+        self._submit_model_timeout = submit_model_timeout
+        self._validation_timeout = validation_timeout
+        self._server_models = server_models
+        self._participating_clients = participating_clients
+        self.num_clients = num_clients
+
+        self._val_results = {}
+        self._client_models = {}
+
+        self._cross_val_models_dir = None
+        self._cross_val_results_dir = None
+
+        self._results_dir = AppConstants.CROSS_VAL_DIR
+        self._json_val_results = {}
+        self._json_file_name = "cross_val_results.json"
+
+    def initialize(self, fl_ctx):
+        super().initialize(fl_ctx)
+
+        # Create shareable dirs for models and results
+        cross_val_path = os.path.join(self.get_run_dir(), self._cross_val_dir)
+        self._cross_val_models_dir = os.path.join(cross_val_path, AppConstants.CROSS_VAL_MODEL_DIR_NAME)
+        self._cross_val_results_dir = os.path.join(cross_val_path, AppConstants.CROSS_VAL_RESULTS_DIR_NAME)
+
+        # Cleanup/create the cross val models and results directories
+        if os.path.exists(self._cross_val_models_dir):
+            shutil.rmtree(self._cross_val_models_dir)
+        if os.path.exists(self._cross_val_results_dir):
+            shutil.rmtree(self._cross_val_results_dir)
+
+        # Recreate new directories.
+        os.makedirs(self._cross_val_models_dir)
+        os.makedirs(self._cross_val_results_dir)
+
+        self._participating_clients = self.sample_clients(self.num_clients)
+
+        for c_name in self._participating_clients:
+            self._client_models[c_name.name] = None
+            self._val_results[c_name.name] = {}
+
+    def run(self) -> None:
+        self.info("Start Cross-Site Evaluation.")
+
+        data = FLModel(params={})
+        data.meta[AppConstants.SUBMIT_MODEL_NAME] = ModelName.BEST_MODEL
+        # Create submit_model task and broadcast to all participating clients
+        self.send_model(
+            task_name=AppConstants.TASK_SUBMIT_MODEL,
+            data=data,
+            targets=self._participating_clients,
+            timeout=self._submit_model_timeout,
+            callback=self._receive_local_model_cb,
+        )
+
+        # Obtain server models and send to clients for validation
+        if self.persistor:
+            for server_model_name in self._server_models:
+                server_model_path = os.path.join(self.get_app_dir(), server_model_name)
+                server_model_learnable = self.persistor.get_model_from_location(server_model_path, self.fl_ctx)
+                server_model = FLModelUtils.from_model_learnable(server_model_learnable)
+                self._send_validation_task(server_model_name, server_model)
+        else:
+            for server_model_name in self._server_models:
+                try:
+                    server_model = fobs.loadf(server_model_name)
```

Comment:
Do we want to add this default model persistor ?

Or we add it in follow up PRs

Reply:
We can add it in follow up PR. One thing to consider is how to specify different save/load locations, as the ModelPersistor doesn't have this parameter for the save/load.